cirq-core/cirq/ops/clifford_gate.py

```diff
@@ -759,13 +759,39 @@ def __pow__(self, exponent: float) -> 'SingleQubitCliffordGate':
 
         return NotImplemented
 
-    def _act_on_(
-        self,
-        sim_state: 'cirq.SimulationStateBase',  # pylint: disable=unused-argument
-        qubits: Sequence['cirq.Qid'],  # pylint: disable=unused-argument
-    ):
-        # TODO(#5256) Add the implementation of _act_on_ with CliffordTableauSimulationState.
-        return NotImplemented
+    def _act_on_(self, sim_state: 'cirq.SimulationStateBase', qubits: Sequence['cirq.Qid']):
+        from cirq.sim.clifford.stabilizer_simulation_state import StabilizerSimulationState
+
+        if not isinstance(sim_state, StabilizerSimulationState):
+            return NotImplemented
+        if self == SingleQubitCliffordGate.I:
+            return True
+        axis = sim_state.get_axes(qubits)[0]
+        stabilizer = sim_state._state
+        match self:
+            case SingleQubitCliffordGate.X:
+                stabilizer.apply_x(axis)
+            case SingleQubitCliffordGate.Y:
+                stabilizer.apply_y(axis)
+            case SingleQubitCliffordGate.Z:
+                stabilizer.apply_z(axis)
+            case SingleQubitCliffordGate.H:
+                stabilizer.apply_h(axis)
+            case SingleQubitCliffordGate.X_sqrt:
+                stabilizer.apply_x(axis, 0.5)
+            case SingleQubitCliffordGate.Y_sqrt:
+                stabilizer.apply_y(axis, 0.5)
+            case SingleQubitCliffordGate.Z_sqrt:
+                stabilizer.apply_z(axis, 0.5)
+            case SingleQubitCliffordGate.X_nsqrt:
+                stabilizer.apply_x(axis, -0.5)
+            case SingleQubitCliffordGate.Y_nsqrt:
+                stabilizer.apply_y(axis, -0.5)
+            case SingleQubitCliffordGate.Z_nsqrt:
+                stabilizer.apply_z(axis, -0.5)
+            case _:
+                return NotImplemented  # Unnamed Cliffords will be decomposed.
```

Comment:
@BichengYing maybe this should fall back to `super()._act_on_()` to be handled by `CliffordGate`? Or maybe there's no need for this override function at all, and the base `CliffordGate` implementation is sufficient?

Reply:
> Or maybe there's no need for this override function at all, and the base `CliffordGate` implementation is sufficient?
The base implementation doesn't handle `ChForm`, so at least we gain that by overriding `CliffordGate._act_on_` here.

But it does seem like here in the unnamed Clifford case, returning `super()._act_on_()` would be a better option than returning `NotImplemented`, which will lead to decomposing. WDYT?

Reply:
Yeah, changed to `super()._act_on_()` after testing performance impact in comment https://github.com/quantumlib/Cirq/pull/7069#issuecomment-2661036052 below.